test/integration/acceptance/transfer_asset_test.cpp

```diff
@@ -0,0 +1,359 @@
+          baseTx().transferAsset(kUser1Id, kUser2Id, kAsset, kDesc, kAmount)))
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .done();
+}
+
+/**
+ * @given some user with only can_transfer permission
+ * @when execute tx with TransferAsset command
+ * @then there is an empty proposal
+ */
+TEST_F(TransferAsset, WithOnlyCanTransferPerm) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms(
+          kUser1, kUser1Keypair, {iroha::model::can_transfer}, kRole1))
+      .sendTx(makeUserWithPerms(kUser2, kUser2Keypair, kPerms, kRole2))
+      .sendTx(addAssets(kUser1, kUser1Keypair))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(baseTx()
+                  .transferAsset(kUser1Id, kUser2Id, kAsset, kDesc, kAmount)
+                  .build()
+                  .signAndAddSignature(kUser1Keypair))
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with only can_receive permission
+ * @when execute tx with TransferAsset command
+ * @then there is an empty proposal
+ */
+TEST_F(TransferAsset, WithOnlyCanReceivePerm) {
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms(
+          kUser1, kUser1Keypair, {iroha::model::can_receive}, kRole1))
+      .sendTx(makeUserWithPerms(kUser2, kUser2Keypair, kPerms, kRole2))
+      .sendTx(addAssets(kUser1, kUser1Keypair))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(baseTx()
+                  .transferAsset(kUser1Id, kUser2Id, kAsset, kDesc, kAmount)
+                  .build()
+                  .signAndAddSignature(kUser1Keypair))
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given some user with all required permissions
+ * @when execute tx with TransferAsset command to inexistent destination
+ * @then there is an empty proposal
+ */
+TEST_F(TransferAsset, InexistentDest) {
+  const std::string &inexistent = "inexist@test"s;
+  IntegrationTestFramework()
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms(kUser1, kUser1Keypair, kPerms, kRole1))
+      .sendTx(addAssets(kUser1, kUser1Keypair))
+      .skipProposal()
+      .skipBlock()
+      .sendTx(baseTx()
+                  .transferAsset(kUser1Id, inexistent, kAsset, kDesc, kAmount)
+                  .build()
+                  .signAndAddSignature(kUser1Keypair))
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given pair of users with all required permissions
+ * @when execute tx with TransferAsset command with inexistent asset
+ * @then there is an empty proposal
+ */
+TEST_F(TransferAsset, InexistentAsset) {
+  const std::string &inexistent = "inexist#test"s;
```

Comment:
I believe you do not need "s literal here

Reply:
it is the same

Reply:
well, godbolt [says](https://godbolt.org/g/7zotAy) it even smaller, but there is no difference at all in release mode anyway

Reply:
Well it is smaller because allocation is move to function operator ""s